sensor/common/scannerclient/grpc_client.go

```diff
@@ -83,10 +176,68 @@ func (c *Client) GetImageAnalysis(ctx context.Context, image *storage.Image, cfg
 
 	log.Debugf("Received image components from local Scanner for image %s", name)
 
-	return resp, nil
+	return &ImageAnalysis{
+		ScanStatus:   resp.GetStatus(),
+		ScanNotes:    resp.GetNotes(),
+		V1Components: resp.GetComponents(),
+	}, nil
+}
+
+// Close closes and cleanup the client connection.
+func (c *v2Client) Close() error {
+	return c.conn.Close()
+}
+
+func convertIndexReportToAnalysis(ir *v4.IndexReport) *ImageAnalysis {
+	var st scannerV1.ScanStatus
+	switch ir.GetState() {
+	case "Terminal", "IndexError":
+		st = scannerV1.ScanStatus_FAILED
+	case "IndexFinished":
+		st = scannerV1.ScanStatus_SUCCEEDED
+	default:
+		st = scannerV1.ScanStatus_ANALYZING
+	}
+	return &ImageAnalysis{
+		ScanStatus: st,
+		V4Contents: ir.GetContents(),
+	}
+}
+
+func (c *v4Client) GetImageAnalysis(ctx context.Context, image *storage.Image, cfg *types.Config) (*ImageAnalysis, error) {
+	hid := fmt.Sprintf("/v4/containerimage/%s", image.GetId())
+	opts := []grpc.CallOption{grpc.WaitForReady(true)}
+	ir, err := c.GetIndexReport(ctx, &v4.GetIndexReportRequest{HashId: hid}, opts...)
+	if err == nil {
+		return convertIndexReportToAnalysis(ir), nil
+	}
+	// Check if index was "not found", so we create.
```

Comment:
Something I need to mention about this is concurrent creates. What if I `GetIndexReport() == codes.NotFound` in two different requests/instances, and they beat each other `CreateIndexReport()`?

From what I understood, ClairCore [will use the underlying DB to wait on manifest by Hash ID](https://github.com/quay/claircore/blob/main/libindex/libindex.go#L186). That means concurrent calls would wait until the index is created or lock timeout.